RPA.MIT.Notification.Function/Notification.cs

```diff
@@ -3,143 +3,146 @@
+                            _logger.LogWarning($"Email status for {result.NotifyId} is 'permanent-failure'");
+                            entity = await _notificationTable.Get(result.PartitionKey, result.RowKey);
                             entity["Status"] = "Permanent Failure";
-                            tableClient.UpdateEntity(entity, ETag.All, TableUpdateMode.Replace);
+                            await _notificationTable.Update(entity);
                             await _eventQueueService.CreateMessage(result.PartitionKey, "permanent-failure", "notification", "Email permanent-failure", result.Data);
                             break;
 
                         case "temporary-failure":
-                            log.LogWarning($"Email status for {result.NotifyId} is 'temporary-failure'");
-                            entity = tableClient.GetEntity<TableEntity>(result.PartitionKey, result.RowKey);
+                            _logger.LogWarning($"Email status for {result.NotifyId} is 'temporary-failure'");
+                            entity = await _notificationTable.Get(result.PartitionKey, result.RowKey);
                             entity["Status"] = "Temporary Failure";
-                            tableClient.UpdateEntity(entity, ETag.All, TableUpdateMode.Replace);
+                            await _notificationTable.Update(entity);
                             await _eventQueueService.CreateMessage(result.PartitionKey, "temporary-failure", "notification", "Email temporary-failure", result.Data);
                             break;
 
                         case "technical-failure":
-                            log.LogWarning($"Email status for {result.NotifyId} is 'technical-failure'");
-                            entity = tableClient.GetEntity<TableEntity>(result.PartitionKey, result.RowKey);
+                            _logger.LogWarning($"Email status for {result.NotifyId} is 'technical-failure'");
+                            entity = await _notificationTable.Get(result.PartitionKey, result.RowKey);
                             entity["Status"] = "Technical Failure";
-                            tableClient.UpdateEntity(entity, ETag.All, TableUpdateMode.Replace);
+                            await _notificationTable.Update(entity);
                             await _eventQueueService.CreateMessage(result.PartitionKey, "technical-failure", "notification", "Email technical-failure", result.Data);
                             break;
 
                         default:
-                            log.LogWarning($"Email status for {result.NotifyId} not found");
+                            _logger.LogWarning($"Email status for {result.NotifyId} not found");
                             await _eventQueueService.CreateMessage(result.PartitionKey, "failed", "notification", "Email status not found", result.Data);
                             break;
                     }
```

Comment:
Just a suggestion, but we could make this more dry/solid with methods such as HandleError(), ParseAndValidateMessage(), SendEmail() etc

Reply:
Code updated to use correct logging.